src/components/stopPoster/adContainer.js

```diff
@@ -21,9 +22,35 @@ class AdContainer extends Component {
     }
 
     componentDidUpdate() {
+        this.updateAds();
         this.updateLayout();
     }
 
+    updateAds() {
```

Comment:
I noticed that the template images were set in the constructor and never updated. The ads seemed to work fine like this, but the map picture did not. Thus I also created a function to update the map images if they change.